lib/widgets/text_input_widget.dart

```diff
@@ -0,0 +1,151 @@
+import 'package:flutter/material.dart';
+import 'package:tech_byte/utils/colors.dart';
+import 'package:tech_byte/utils/constants.dart';
+
+class TBTextInput extends StatefulWidget {
+  final TextEditingController textEditingController;
+  final String label;
+  final String? Function(String?)? validator;
+  final bool? enabled;
+  const TBTextInput(
+      {super.key,
+      required this.textEditingController,
+      required this.label,
+      this.validator,
+      this.enabled});
+
+  @override
+  State<TBTextInput> createState() => _TBTextInputState();
+}
+
+class _TBTextInputState extends State<TBTextInput> {
+  final FocusNode _focusNode = FocusNode();
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return TextFormField(
+      enabled: widget.enabled,
+      onTapOutside: (event) {
+        _focusNode.unfocus();
+      },
+      validator: widget.validator,
+      controller: widget.textEditingController,
+      focusNode: _focusNode,
+      style: MaterialStateTextStyle.resolveWith(
+        (states) {
+          if (states.contains(MaterialState.disabled)) {
+            return TextStyle(
+                color: TBColor.textInput.grey,
+                fontFamily: "Inter",
+                fontSize: 16,
+                fontWeight: FontWeight.w600);
+          }
+
+          return TextStyle(
+              color: TBColor.textInput.black,
+              fontFamily: "Inter",
+              fontSize: 16,
+              fontWeight: FontWeight.w600);
+        },
+      ),
```

Comment:
Bad coding, code like in the image and go through whole code and refactor all places where the state is using
![image](https://github.com/damjan-Igrowth/INT-Milan_Zezelj-Shop/assets/119666343/b82ae8ce-520a-4b38-ab6e-8725789c5290)